rustworkx-core/src/generators/simple_shortest_paths.rs

```diff
@@ -0,0 +1,368 @@
+                simple_graph = &self.unique_paths[self.switch];
+                self.index = 0;
+                index = 0;
+            } else {
+                return None;
+            }
+        }
+
+        let edge = self.graph.find_edge(simple_graph[index], simple_graph[index + 1]);
+        match edge {
+            Some(edge) => {
+                let (s, t) = (simple_graph[index], simple_graph[index + 1]);
+                let Some(weight) = self.graph.edge_weight(edge) else {
+                    return None;
+                };
+                let weight = *weight;
+                
+                println!("Removing edge {:?} {:?}",s,t);
+                {
+                let graph = &mut self.graph;
+                graph.remove_edge(edge);
+                }
+
+                let sim_path = get_simple_path(self);
+
+
+                
+
+                match sim_path {
+                    None => {
+                        self.index = self.index + 1;
+                        let graph = &mut self.graph;
+                        graph.add_edge(s, t, weight);
+                        return self.next();
+                    }
+                    Some(mut s_path) => {
+                        {   
+                            let graph = &mut s_path.graph;
+                            graph.add_edge(s, t, weight);
+                        }
+                        return Some(s_path);
+                    }
+                }
+            }
+            None => {
+                self.index = self.index + 1;
+                return self.next();
+            }
+        }
+    }
+}
+// The code provides the shortest distance cost of all Nodes from the start Node
+#[derive(Copy, Clone, Debug)]
+struct MinScored<K, T>(pub K, pub T);
+
+impl<K: PartialOrd, T> PartialEq for MinScored<K, T> {
+    #[inline]
+    fn eq(&self, other: &MinScored<K, T>) -> bool {
+        self.cmp(other) == Ordering::Equal
+    }
+}
+
+impl<K: PartialOrd, T> Eq for MinScored<K, T> {}
+
+impl<K: PartialOrd, T> PartialOrd for MinScored<K, T> {
+    #[inline]
+    fn partial_cmp(&self, other: &MinScored<K, T>) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl<K: PartialOrd, T> Ord for MinScored<K, T> {
+    #[inline]
+    fn cmp(&self, other: &MinScored<K, T>) -> Ordering {
+        let a = &self.0;
+        let b = &other.0;
+        if a == b {
+            Ordering::Equal
+        } else if a < b {
+            Ordering::Greater
+        } else if a > b {
+            Ordering::Less
+        } else if a.ne(a) && b.ne(b) {
+            // these are the NaN cases
+            Ordering::Equal
+        } else if a.ne(a) {
+            // Order NaN less, so that it is last in the MinScore order
+            Ordering::Less
+        } else {
+            Ordering::Greater
+        }
+    }
+}
+
+// This is mutation of petgraph dijkastra to get full path between source to target
+fn dijkstra<G, F, K>(
```

Comment:
I have not review this specific Dijkastra, but petgraph dijkastra returns only score of each node between source & target. but how to find the path from source -> target is not clear from that score. thats why i have updated the code to return the previous node as, that it followed to reach to the destination.
like - if the path for lowest score was A->B-->C->D, it will return the score, also Node (C), which is the previous node, similarly for C , it will return score & B as previous node, and for B it will return A as previous node, which we can use to reach till A from D.

Let me know if I am missing something on existing Dijkastra functioning.